File: frontend/src/game/upgrades/curses/damage_reduc_2.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const DamageReduc2Def: UpgradeDef = {
  id: "damage_reduc_2",
  name: "Weakness 2",
  description: "-0.4% damage.",
  rarity: RarityID.Uncommon,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 0,
  targetClass: UpgradeTargetID.Attack,
  fieldInTargetClass: UpgradeStatID.Damage,
  value: -0.004,
  isMultiplier: true,
  stackable: true,
  maxStacks: 99999,
  curse: true,
}

export class DamageReduc2 extends Upgrade {}

File: backend/app/core/upgrades/curses/damage_reduc_2.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class DamageReduc2(UpgradeImplementation):
    id = 'damage_reduc_2'
    name = 'Weakness 2'
    description = '-0.4% damage.'
    rarity = 'uncommon'
    type = 'stat_modifier'
    cost = 0

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = -0.004
        appliers.modifiers.add_multiplier('attack', 'damage', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True
